lib/draper/automatic_delegation.rb

```diff
@@ -6,8 +6,7 @@ module AutomaticDelegation
     def method_missing(method, *args, &block)
       return super unless delegatable?(method)
 
-      self.singleton_class.delegate method
-      send(method, *args, &block)
+      object.send(method, *args, &block)
```

Comment:
Thank you for trying :)
I understand difficulty of it. ([Kernel#caller](https://docs.ruby-lang.org/en/2.4.0/Kernel.html#method-i-caller) might be able to do. But I think it  is too much supporting)

I agree your suggestion. It looks ok if that behavior covered by black box test. (and has documentation 👍 )